osquery/tables/system/darwin/signature.mm

```diff
@@ -91,92 +91,95 @@ void genSignatureForFile(const std::string& path, QueryData& results) {
   result =
       SecStaticCodeCheckValidityWithErrors(staticCode, flags, nullptr, nullptr);
   if (result == errSecSuccess) {
-    CFDictionaryRef codeInfo = nullptr;
-
-    result = SecCodeCopySigningInformation(
-        staticCode, kSecCSSigningInformation | kSecCSRequirementInformation,
-        &codeInfo);
-    if (result == errSecSuccess) {
-      // If we don't get an identifier for this file, then it's not signed.
-      CFStringRef ident =
-          (CFStringRef)CFDictionaryGetValue(codeInfo, kSecCodeInfoIdentifier);
-      if (ident != nullptr) {
-        // We have an identifier - this indicates that the file is signed,
-        // and, since it didn't error above, it's *also* a valid signature.
+    isValidated = true;
+  } else {
+    // If this errors, then we either don't have a signature, or it's malformed.
+    VLOG(1) << "Static code validity check failed for file: " << path;
+  }
+  CFDictionaryRef codeInfo = nullptr;
+  result = SecCodeCopySigningInformation(
+      staticCode, kSecCSSigningInformation | kSecCSRequirementInformation,
+      &codeInfo);
+  if (result == errSecSuccess) {
+    // If we don't get an identifier for this file, then it's not signed.
+    CFStringRef ident =
+        (CFStringRef)CFDictionaryGetValue(codeInfo, kSecCodeInfoIdentifier);
+    if (ident != nullptr) {
+      // We have an identifier - this indicates that the file is signed,
+      // and, since it didn't error above, it's *also* a valid signature.
+      if (isValidated) {
         r["signed"] = INTEGER(1);
-        r["identifier"] = stringFromCFString(ident);
-
-        // Get CDHash
-        r["cdhash"] = "";
-        CFTypeRef hashInfo = CFDictionaryGetValue(codeInfo, kSecCodeInfoUnique);
-        if (hashInfo != nullptr) {
-          r["cdhash"].reserve(CC_SHA1_DIGEST_LENGTH);
-          // Get the SHA-1 bytes
-          std::stringstream ss;
-          auto bytes = CFDataGetBytePtr((CFDataRef)hashInfo);
-          if (bytes != nullptr) {
-            // Write bytes as hex strings
-            for (unsigned n = 0; n < CC_SHA1_DIGEST_LENGTH; n++) {
-              ss << std::hex << std::setfill('0') << std::setw(2);
-              ss << (unsigned int)bytes[n];
-            }
-            r["cdhash"] = ss.str();
-          }
-          if (r["cdhash"].length() != CC_SHA1_DIGEST_LENGTH * 2) {
-            VLOG(1) << "Error extracting code directory hash";
-            r["cdhash"] = "";
+      }
+      r["identifier"] = stringFromCFString(ident);
+
+      // Get CDHash
+      r["cdhash"] = "";
+      CFTypeRef hashInfo = CFDictionaryGetValue(codeInfo, kSecCodeInfoUnique);
+      if (hashInfo != nullptr) {
+        r["cdhash"].reserve(CC_SHA1_DIGEST_LENGTH);
+        // Get the SHA-1 bytes
+        std::stringstream ss;
+        auto bytes = CFDataGetBytePtr((CFDataRef)hashInfo);
+        if (bytes != nullptr) {
+          // Write bytes as hex strings
+          for (unsigned n = 0; n < CC_SHA1_DIGEST_LENGTH; n++) {
```

Comment:
Prefer `size_t` as the loop counter type.

I think you also need to check if `CFDataGetLength(bytes) < CC_SHA1_DIGEST_LENGTH`, otherwise could be accessing have out of bounds.